index.js

```diff
@@ -99,12 +99,32 @@ module.exports = {
 
     // We shouldn't enforce filename casing to always be the same.  e.g. We may
     // prefer kebab-case for bin files, PascalCase for React components, etc.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/filename-case.md
     'unicorn/filename-case': 'off',
 
     // Uses safe-regex to disallow potentially catastrophic exponential-time
     // regular expressions.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-unsafe-regex.md
     'unicorn/no-unsafe-regex': 'error',
 
+    // This rule is onerous in React functional components, has been debated
+    // by the community and proven as an unnecessary optimization
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/consistent-function-scoping.md
+    'unicorn/consistent-function-scoping': 'off',
+
+    // This rule is incompatible with React
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-null.md
+    'unicorn/no-null': 'off',
+
+    // Preventing abbreviations is too opinionated for linting and is not
+    // particularly compatible with React development (props, ref, etc.)
```

Comment:
Could the (intended) benefits of the rule still be gained by putting the React nomenclature into the [`whitelist`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prevent-abbreviations.md#whitelist) configuration rather than fully disabling?

Reply:
No because there are lots of other non-React places where abbreviations are fine.

I disagree philosophically with this rule that abbreviations are inherently bad or unreadable. It's too nitpicky.